tests/flytekit/unit/core/test_type_engine.py

```diff
@@ -1794,6 +1802,9 @@ def test_union_of_lists():
 def test_list_of_unions():
```

Comment:
Should we run this test only when Python version >= 3.10?

```python
@pytest.mark.skipif(sys.version_info < (3, 10))
def test_optional_type():
  ...
```

Reply:
We have another cron job that runs every day. https://github.com/flyteorg/flytekit/blob/a150899c8ac8b49cc7fed28c8771a2c0039f9681/.github/workflows/pythonbuild.yml#L11

Reply:
gtk, yeah I can skip all of the test with `|` syntax